src/transports/http.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#ifndef INCLUDE_http_h__
+#define INCLUDE_http_h__
+
+#include "buffer.h"
+
+#define GIT_HTTP_USER_AGENT_FMT "git/2.0 (%s)"
+
+GIT_INLINE(const char *) git_http__user_agent(void)
+{
+	const char *custom = git_libgit2__user_agent();
+
+	if (custom)
+		return custom;
+
+	return "libgit2 " LIBGIT2_VERSION;
+}
```

Comment:
Why is the definition inline and not part of `http.c`?

Reply:
The `http.c` and `winhttp.c` files are independent from each other.

Reply:
Yeah. It feels a bit awkward that "winhttp.c" refers to "http.h", but the headers complement "http.c" does in fact have nothing to do with "winhttp.c". I mean we could rename "http.c" to "unixhttp.c" and have "http.{c,h}" contain common code to both. But I'm definitly not keen on "unixhttp.c", and for the lack of a better name I'd just go with your proposal.